src/main/java/org/folio/inventory/resources/InstancesBatch.java

```diff
@@ -226,11 +227,11 @@ private void setInstancesIdIfNecessary(List<JsonObject> instances) {
             });
         }
       }
-      CompositeFuture.join(updateRelationshipsFutures).setHandler(resultFuture);
+      CompositeFuture.join(updateRelationshipsFutures).onComplete(resultFuture);
     } catch (IllegalStateException e) {
       log.error("Can not update instances relationships cause: " + e);
       resultFuture.fail(e);
     }
-    return resultFuture;
+    return resultFuture.future();
```

Comment:
There is no need for a promise, we can directly return the Future:
`return CompositeFuture.join(updateRelationshipsFutures);`
`return Future.failedFuture(e);`